src/main/java/uk/co/bconline/ndelius/service/impl/UserServiceImpl.java

```diff
@@ -109,11 +115,19 @@ public UserServiceImpl(
 		val dbFuture = supplyAsync(() -> userEntityService.search(query, includeInactiveUsers, datasetFilter), taskExecutor);
 		val ldapFuture = supplyAsync(() -> userEntryService.search(query, includeInactiveUsers, datasetFilter), taskExecutor);
 
+		val roleFuture = supplyAsync(() -> roleService.getUsersRoles(role).stream()
+				.map(user -> user.getDn().get(user.getDn().size() - 2).split("cn=")[1]) // username is 2nd-to-last part of distinguished name
```

Comment:
Could we use `LdapUtils::getStringValue` for this? Should be less flaky than manually splitting the string. Think it would be something like:
```suggestion
				.map(user -> LdapUtils.getStringValue(user.getDn(), user.getDn().size() - 2)) // username is 2nd-to-last part of distinguished name
```
or maybe this (not sure which cn value this would retrieve):
```suggestion
				.map(user -> LdapUtils.getStringValue(user.getDn(), "cn"))
```

Reply:
With a dn like `cn=UMBT001,cn=test.gdpr.client,cn=EISUsers,ou=Users`, the second solution would return `EISUsers` rather than `test.gdpr.client`, so I think we'll have to go with `LdapUtils.getStringValue(user.getDn(), user.getDn().size() - 2)`